src/provider/provider_os_memory.c

```diff
@@ -337,7 +337,8 @@ static umf_result_t os_free(void *provider, void *ptr, size_t size) {
 
     errno = 0;
     int ret = os_munmap(ptr, size);
-    if (ret) {
+    // ignore error when size == 0
```

Comment:
Do we have a test for calling umfMemoryProviderFree with size 0?

Reply:
Test added.
Done.